dbms/src/Columns/IColumn.h

```diff
@@ -345,6 +355,9 @@ class IColumn : public COWPtr<IColumn>
 
     /// Different from scatter, scatterTo appends the scattered data to 'columns' instead of creating ScatterColumns
     virtual void scatterTo(ScatterColumns & columns, const Selector & selector) const = 0;
+    virtual void scatterTo(ScatterColumns & columns, const Selector & selector, const BlockSelectivePtr & selective_ptr)
```

Comment:
Better not to use a `shared_ptr` parameter for a common `IColumn` function because it will force other callers to allocate a `shared_ptr`.
How about
```
using Selectives = PaddedPODArray<UInt64>;
virtual void scatterTo(ScatterColumns & columns, const Selector & selector, const Selectives & selectives) const = 0;
```

Reply:
Normally selective info is passed by shared_ptr

Reply:
done